File: akka-docs/src/test/scala/docs/stream/operators/sourceorflow/LimitWeighted.scala
/*
 * Copyright (C) 2009-2023 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.stream.operators.sourceorflow

import akka.NotUsed
import akka.actor.typed.ActorSystem
import akka.stream.scaladsl.Source
import akka.util.ByteString

import scala.annotation.nowarn
import scala.concurrent.Future

@nowarn("msg=never used") // sample snippets
object LimitWeighted {

  implicit val system: ActorSystem[_] = ???

  def simple(): Unit = {
    // #simple
    val untrustedSource: Source[ByteString, NotUsed] = Source.repeat(ByteString("element"))

    val allBytes: Future[ByteString] =
      untrustedSource.limitWeighted(max = 10000)(_.length).runReduce(_ ++ _)
    // #simple
  }

}

File: akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/LimitWeighted.java
/*
 * Copyright (C) 2009-2022 Lightbend Inc. <https://www.lightbend.com>
 */

package jdocs.stream.operators.sourceorflow;

import akka.NotUsed;
import akka.actor.typed.ActorSystem;
import akka.stream.javadsl.Source;
import akka.util.ByteString;
import java.util.concurrent.CompletionStage;

public class LimitWeighted {
  public void simple() {
    ActorSystem<?> system = null;
    // #simple
    Source<ByteString, NotUsed> untrustedSource = Source.repeat(ByteString.fromString("element"));

    CompletionStage<ByteString> allBytes =
        untrustedSource
            .limitWeighted(
                10000, // max bytes
                bytes -> (long) bytes.length() // bytes of each chunk
                )
            .runReduce(ByteString::concat, system);
    // #simple
  }
}
